File: backend/src/main/java/com/slior/dto/route/RouteResponse.java
package com.slior.dto.route;

import com.slior.model.Route;
import com.slior.model.enums.RouteStatus;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record RouteResponse(
        UUID id,
        String nombre,
        LocalDate fechaPlanificada,
        RouteStatus status,
        UUID repartidorId,
        String repartidorNombre,
        List<StopResponse> paradas,
        Double distanciaTotal,
        Integer tiempoEstimado,
        String notas,
        LocalDateTime createdAt
) {
    public static RouteResponse from(Route route) {
        return new RouteResponse(
                route.getId(),
                route.getNombre(),
                route.getFechaPlanificada(),
                route.getStatus(),
                route.getRepartidor().getId(),
                route.getRepartidor().getNombre(),
                route.getStops().stream()
                        .filter(s -> !s.isDeleted())
                        .map(StopResponse::from)
                        .toList(),
                route.getDistanciaTotal(),
                route.getTiempoEstimado(),
                route.getNotas(),
                route.getCreatedAt()
        );
    }
}

File: mobile-app/app/src/main/java/com/slior/data/remote/dto/RouteResponse.kt
package com.slior.data.remote.dto

import com.google.gson.annotations.SerializedName

data class StopResponseDto(
    val id: String,
    val direccion: String,
    val destinatario: String,
    val telefonoDestinatario: String,
    val latitud: Double,
    val longitud: Double,
    val ordenVisita: Int,
    val status: String,
    val notas: String?,
    val entregadoEn: String?
)

data class RouteResponseDto(
    val id: String,
    val nombre: String,
    val fechaPlanificada: String,
    val status: String,
    val repartidorId: String,
    val repartidorNombre: String,
    val paradas: List<StopResponseDto>,
    val distanciaTotal: Double?,
    val notas: String?,
    val createdAt: String,
    val tiempoEstimado: Int?
)